powerstore/datasource_snapshotrule_test.go

```diff
@@ -0,0 +1,86 @@
+package powerstore
+
+import (
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+// Test to Fetch SnapshotRule
+func TestAccSnapshotRule_FetchSnapshotRule(t *testing.T) {
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("Dont run with units tests because it will try to create the context")
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testProviderFactory,
+		Steps: []resource.TestStep{
```

Comment:
negative scenarios can also be added . Also scenarios with empty id / name

Reply:
Added